### PR TITLE
Readds secborgs (as a traitor item)

### PR DESCRIPTION
**Pull request by InsaneHyena**
:cl: 
add: Added a roboticist-specific traitor item - a box of Bloodhound upgrade boards.
/:cl:

Adds a board that can hack cyborgs and turn them into secborgs. Babby's first code that likely doesn't even work.

### CODE REVIEW

**Comment by coiax** [comment]
FEAR THE BLOODHOUND

**Comment by InsaneHyena** [comment]
That's a good point, maybe it should be called Bloodhound upgrade module instead.

**Comment by PKPenguin321** [comment]
Hmm. I was skeptical of this on the forums because I assumed it would be infinite use, but one secborg might not be as bad. 

Either way, I'm extremely cautious of this, secborgs are really really strong and since they're a hard traitor tell now they'll be used exclusively to murderbone, and if they can take out the ion rifle/robotics console(s) and have a human to stick with them they're basically unstoppable. Overall a :-1: from me, but not as strong as I was initially going to give it.

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 1]
```diff
@@ -381,3 +381,22 @@
...
+	return
+
+/obj/item/borg/upgrade/afterattack(atom/target, mob/user, proximity)
+	var/atom/A = target
+	if(!proximity && prox_check)
+		return
+	if(!isrobot(target))
+		return
+	A.emag_act(user)
+	A.transform to(/obj/item/weapon/robot_module/security)
```
Comment (RemieRichards) [inline code/game/objects/items/robot/robot_upgrades.dm]: _

File: code/modules/uplink/uplink_item.dm [hunk 2]
```diff
@@ -1223,6 +1223,14 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
 	cost = 20
 	surplus = 0
 	restricted_roles = list("Assistant")
+	
+/datum/uplink_item/role_restricted/traitorsec
+	name = "O.V.E.R.S.E.E.R. upgrade module"
+	desc = "An illegal cyborg upgrade module that both hacks the cyborg and activates it's hidden security module."
```
Comment (RemieRichards) [inline code/modules/uplink/uplink_item.dm]: its

File: code/game/objects/items/robot/robot_upgrades.dm (hunk 1 above)
Comment (InsaneHyena) [inline code/game/objects/items/robot/robot_upgrades.dm]: What?

Comment (RemieRichards) [inline code/game/objects/items/robot/robot_upgrades.dm]: notice your code doesn't work, then notice what I posted.

**Comment by InsaneHyena** [comment]
It being one use wasn't actually my intention. I think it should be two or three uses, so I'll work on that.

**Comment by Supermichael777** [comment]
obviously the order is hack > console and spare > ion > wipe rd > upgrade

**Comment by anconfuzedrock** [comment]
I'm gonna be honest, the main reason i didnt do this myself is because I thought people would hate it and it was too complicated and I am too stupid to do this, good on ya cuz I didnt know how to do this myself.

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 3]
```diff
@@ -381,3 +381,22 @@
...
+	name = "Bloodhound upgrade module"
+	desc = "Ilegal upgrade board that unlocks the banned security module"
+	icon_state = "cyborg_upgrade3"
+	require_module = 1
+	origin_tech = "data=4;combat=4;syndicate=3"
+	
+/obj/item/borg/upgrade/traitorsec/attack()
+	return
+
+/obj/item/borg/upgrade/afterattack(atom/target, mob/user, proximity)
```
Comment (Cyberboss) [inline code/game/objects/items/robot/robot_upgrades.dm]: Is no one going to comment on the fact that this is a one click transform??

**Comment by InsaneHyena** [comment]
I have no idea what I'm doing.

**Comment by Kraseo** [comment]
try merging with upstream

**Comment by Quardbreak** [comment]
From master branch, aaaaaaaaaaaaaaaaa

**Comment by InsaneHyena** [comment]
I actually intended to include all of those commits. I'm referring to the code itself.

**Comment by ShadowDeath6** [comment]
for future reference, don't directly do stuff from your master branch, it gets irritating to clean up when you want to make different prs

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 4]
```diff
@@ -381,3 +381,20 @@
 	R.module.add_module(S, FALSE, TRUE)
 
 	return 1
+	
+/obj/item/borg/upgrade/traitorsec
+        name = "Bloodhound upgrade module"
```
Comment (JJRcop) [inline code/game/objects/items/robot/robot_upgrades.dm]: All of these spaces should be tabs, this tab should only have one indentation.

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 5]
```diff
@@ -381,3 +381,20 @@
...
+        icon_state = "cyborg_upgrade3"
+        require_module = 1
+        origin_tech = "data=4;combat=4;syndicate=3"
+
+
+/obj/item/borg/upgrade/traitorsec/action(mob/living/silicon/robot/R,/mob/living/user)
+	if(..())
+		return 0
+       
+       wiresexposed = TRUE
```
Comment (JJRcop) [inline code/game/objects/items/robot/robot_upgrades.dm]: All of these lines should have only one level of indentation, and these wiresexposed lines should be R.wiresexposed

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 6]
```diff
@@ -381,3 +381,20 @@
...
+        A.emag_act(user)
```
Comment (JJRcop) [inline code/game/objects/items/robot/robot_upgrades.dm]: `A` should be `R`

File: code/game/objects/items/weapons/storage/uplink_kits.dm [hunk 7]
```diff
@@ -315,4 +315,11 @@
 /obj/item/weapon/storage/box/syndie_kit/mimery/New()
 	..()
 	new /obj/item/weapon/spellbook/oneuse/mimery_blockade(src)
-	new /obj/item/weapon/spellbook/oneuse/mimery_guns(src)
\ No newline at end of file
+	new /obj/item/weapon/spellbook/oneuse/mimery_guns(src)
+	
+	/obj/item/weapon/storage/box/syndie_kit/traitorsec_boards/New()
```
Comment (JJRcop) [inline code/game/objects/items/weapons/storage/uplink_kits.dm]: This should have no identation, remove this tab.

**Review by JJRcop — CHANGES_REQUESTED** [review]
Please fix the indentation issues, and not properly using proc arguments.

File: code/game/objects/items/robot/robot_upgrades.dm (hunk 5 above)
Comment (JJRcop) [inline code/game/objects/items/robot/robot_upgrades.dm]: There is an exception in the `return 0` line, it should have two levels of indentation.

**Comment by InsaneHyena** [comment]
I thank all of those brave souls (Antur, Haev, JJRCop) who helped me with this trash code.

**Comment by ExcessiveUseOfCobblestone** [comment]
If you make them one use, you can make them 7 a piece, meaning 1) you'll only get 2 unless you can befriend a traitor bro [see: kill them with PDA/Radio/Pen unlocked] and 2) you'll have /just enough/ [intentional] to get an emag but nothing else because you have 2 killing machines.

Tada, B A L A N C E

**Comment by PKPenguin321** [comment]
>now it has three

Even two is enough to hijack, I know this by experience. One was already too much if you ask me because you have to remember that not only is the secborg already ridiculous, it also has a guaranteed teammate (the guy that hacked it) to repair it, drag it from stuns, change its battery, etc. and vice versa

I'm serious when I say this isn't a good idea, if you thought dual eswords were too good for murderbone you'll hate secborgs

**Comment by InsaneHyena** [comment]
I disagree, since we're talking about borgs who are essentially giant VALID signs. But I'm open to idea of 7tc for one board.

**Comment by Incomptinence** [comment]
PKP i find it making hijack super easy unlikely. As soon as the bloodhounds are out the traitor is super loud. If they get spotted it is going to be malf level panic mode without a ton of set up using regular borgs to nab ai dissable coms etc. Slightly better finisher perhaps.

Now onto how much of a facetious pile of excrement you are my dear penguin. Double eswords provide exceptional protection against all manner of damaging projectiles. Sec borg has defense against no damaging projectiles.

**Comment by PKPenguin321** [comment]
I'm sure I've told this story a thousand times before of the time I hijacked as traitor robo two rounds in a row on high pop as traitor robo with actually 0 subtlety involved. It doesn't matter if you know the secborg is a traitor, it's too strong for you to do anything about it. When I hijacked twice, I didn't even have to close thr robotics shutters, and everybody on the radio kept saying I was killing everybody and to just avoid robotics. One secborg and I were able to take out two sec officers plus the captain with his laser gun plus the HoS with the ion rifle all at the same time.

As for your stupid comment at the end there, secborgs actually do have a defense against that. It's called moving away and shooting, or just closing and bolting doors behind them. They have just as much range as anybody else with a gun, and don't take slowdown from damage, so even if you somehow shoot them to 1 health before they just tase you or whatever they can just go through a door that you can't and go back to their roboticist to get healed (this is all assuming that there isn't already a roboticist nearby to fight you while you're shooting the borg). Again, I am talking from experience. I don't know why you're in such denial.

**Comment by InsaneHyena** [comment]
If bloodhounds prove to be too strong/too weak, we can always adjust the price later. Are you fine with 7tc for one board as a compromise for now?

**Comment by PKPenguin321** [comment]
That is probably worse since you could stack it with an ebow

**Comment by ExcessiveUseOfCobblestone** [comment]
To cull the fears, it will probably be best to make it 14. I rather it be garbo first, buffed later than OH GOD SAVE ME first, nerfed to uselessness right after implementation.

**Comment by InsaneHyena** [comment]
Perfectly valid point.

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 8]
```diff
@@ -381,3 +381,20 @@
...
+	icon_state = "cyborg_upgrade3"
+	require_module = 1
+	origin_tech = "data=4;combat=4;syndicate=3"
+
+
+/obj/item/borg/upgrade/traitorsec/action(mob/living/silicon/robot/R,mob/living/user)
+	if(..())
+		return 0
+    
+	R.wiresexposed = TRUE
```
Comment (GunHog) [inline code/game/objects/items/robot/robot_upgrades.dm]: Remove this. The borg's panel must be open in order for you to insert the upgrade in the first place, and `wiresexposed` is incorrect anyway.

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 9]
```diff
@@ -381,3 +381,20 @@
...
+	R.emag_act(user)
+	R.wiresexposed = FALSE
```
Comment (GunHog) [inline code/game/objects/items/robot/robot_upgrades.dm]: Remove this one as well. Not needed.

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 10]
```diff
@@ -381,3 +381,20 @@
...
+	R.module.transform_to(/obj/item/weapon/robot_module/security)
```
Comment (GunHog) [inline code/game/objects/items/robot/robot_upgrades.dm]: Put this **BEFORE** the `emag_act()`.

**Review by GunHog — CHANGES_REQUESTED** [review]
See my line comments. Remove the changes to the wires var, and swap the order of the module change and emagging.

**Comment by GunHog** [comment]
I have taken the liberty of testing this for you (Because I like the feature and want to see it added). Please make the attempt to test your code before you upload it. Never touch the github web editor, you should have Git set up on your computer so you can compile and run your own code yourself.

**Comment by PKPenguin321** [comment]
To people who want this: Why?

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 11]
```diff
@@ -381,3 +381,19 @@
...
+	require_module = 1
+	origin_tech = "data=4;combat=4;syndicate=3"
+
+
+/obj/item/borg/upgrade/traitorsec/action(mob/living/silicon/robot/R,mob/living/user)
+	if(..())
+		return 0
+		
+	R.module.transform_to(/obj/item/weapon/robot_module/security)
+	R.emag_act(user)
```
Comment (ChangelingRain) [inline code/game/objects/items/robot/robot_upgrades.dm]: you need to also return 1 after this

**Review by GunHog — APPROVED** [review]

File: code/game/objects/items/robot/robot_upgrades.dm (hunk 11 above)
Comment (GunHog) [inline code/game/objects/items/robot/robot_upgrades.dm]: Oh well, I cannot catch *everything*!

**Comment by Iamgoofball** [comment]
1. Lock it to 1, please.
2. Make sure it's a proper syndieborg and not a normal secborg. If it's a normal secborg this will just become the new "friendly traitor lol" meme or the new "HoS-but-with-traitor-gear" setup ala holoparas

**Comment by Anonmare** [comment]
Give it the Syndie black paintjob so it's an immediately obvious "Valid me" sign, even to players off the server.

**Comment by InsaneHyena** [comment]
I think it can be done, but I'm not a spriter. Should I just use the syndieborg sprites? In that case, wouldn't a whole new cyborg module almost identical to security be created? I'll try to figure it out.

**Comment by AdamElTablawy** [comment]
Meanwhile on PKP circlejerks about how overpowered a cyborg any other server can deal with is :^)

**Comment by PKPenguin321** [comment]
Meanwhile on "not one person has given a good reason for adding this," not one person has given a good reason for adding this

**Comment by InsaneHyena** [comment]
Okay, I'm done. I've tested it, and it works perfectly. Had to make bloodhounds a daughter module to secborgs in order to make animations look and work nice.

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 12]
```diff
@@ -384,7 +384,7 @@
 	
 /obj/item/borg/upgrade/traitorsec
 	name = "Bloodhound upgrade module"
-	desc = "Ilegal upgrade board that unlocks the banned security module"
+	desc = "Ilegal upgrade board that unlocks banned security module"
```
Comment (Iamgoofball) [inline code/game/objects/items/robot/robot_upgrades.dm]: :+1:

Comment (PKPenguin321) [inline code/game/objects/items/robot/robot_upgrades.dm]: fix this and also put punctuation

**Review by JJRcop — COMMENTED** [review]
The code is fine now but I don't approve or disapprove of the concept.

**Comment by optimumtact** [comment]
This is on hold until I can ask the headmins for their opinion because this sidesteps the config that disables secborgs.

**Comment by PKPenguin321** [comment]
Not one person has said why they want this aside from one guy memeing about spiting me

**Comment by Ausops** [comment]
I'd be willing to testmerge this if the price were increased. Also try re-skinning it into syndiborg-lite or the '''''wargiver''''''' borg and I bet no one will ever screech about secborgs.

**Comment by InsaneHyena** [comment]
> If the price was increased
Fine, I guess.
> Reskin it into syndiborg-lite
I already did.

**Comment by Anonmare** [comment]
@PKPenguin321 Because it might be fun

**Comment by optimumtact** [comment]
This is better left for a full seccborg verhaul if such a thing can even work

**Pull request closed without merging**